plugins/org.komodo.core/src/org/komodo/repository/search/ObjectSearcher.java

```diff
@@ -0,0 +1,433 @@
+        ArgCheck.isTrue(keywords.length > 0, "Where Containts clause requires at least 1 value"); //$NON-NLS-1$
+
+        alias = checkWhereAlias(alias);
+
+        if (whereClauses == null)
+            whereClauses = new ArrayList<Clause>();
+
+        ContainsClause whereClause = findWhereContainsClause(alias, property);
+        if (whereClause == null) {
+            whereClause = new ContainsClause(alias, property, keywordCriteria, keywords);
+            whereClauses.add(whereClause);
+        }
+        else {
+            for (String keyword : keywords)
+                whereClause.addKeyword(keyword);
+        }
+    }
+
+    /**
+     * Add a CONTAINS sub-clause to the WHERE clause, eg. WHERE CONTAINS (alias.property, 'value')
+     *
+     * @param operator the AND/OR operator preceding the clause. Can be <null> if the first clause
+     * @param alias the alias of the type
+     * @param property the name of the property belonging to the type
+     * @param keyword the keyword to search for in the property
+     */
+    public void addWhereContainsClause(LogicalOperator operator, String alias, String property, String keyword) {
+        addWhereContainsClause(operator, alias, property, KeywordCriteria.ANY, keyword);
+    }
+
+    /**
+     * Add a PATH clause to the Where clause, eg. WHERE PATH() IN (path1, path2)
+     *
+     * @param operator the AND/OR operator preceding the clause. Can be <null> if the first clause
+     * @param paths the paths to be added
+     */
+    public void addWherePath(LogicalOperator operator, String... paths) {
+        ArgCheck.isNotNull(paths);
+        ArgCheck.isTrue(! getFromTypes().isEmpty(), "Where Path clause requires at least 1 From clause"); //$NON-NLS-1$
+        ArgCheck.isTrue(paths.length > 0, "Where Path clause requires at least 1 value"); //$NON-NLS-1$
+
+        if (whereClauses == null)
+            whereClauses = new ArrayList<Clause>();
+
+        PathClause pathClause = new PathClause(paths);
+        whereClauses.add(pathClause);
+    }
+
+    /**
+     * Add a PATH clause to the Where clause, eg. WHERE PATH() IN (path1, path2)
+     *
+     * @param paths the paths to be added
+     */
+    public void addWherePath(String... paths) {
+       addWherePath(null, paths); 
+    }
+
+    /**
+     * Set a custom where clause for this searcher. Such a clause may be created
+     * using {@link #toString()} or be completely custom.
+     * 
+     * Note:
+     * * no parsing of this clause will be conducted prior to appending this clause.
+     * * do not prefix this clause with the WHERE keyword
+     *
+     * @param whereClause custom where clause string
+     */
+    public void setCustomWhereClause(String whereClause) {
+        ArgCheck.isNotEmpty(whereClause);
+        this.customWhereClause = whereClause;
+    }
+
+    private void createSelect(StringBuffer buffer) {
```

Comment:
Why are we selecting all properties every time? Should we have another clause that allows a subset of properties to be selected? Or maybe just hard code `path` and `index` into the select statement since that is all we use to create Komodo objects.